src/main/java/com/redfin/sitemapgenerator/SitemapGenerator.java

```diff
@@ -163,6 +165,38 @@ THIS getThis() {
 		return outFiles;
 	}
 	
+	/**
+	 * Writes out the sitemaps as strings
+	 * @return a list of XML-formatted strings
+	 */
+	public List<String> writeAsString() {
+		List<String> listOfSiteMapStrings = new ArrayList<String>();
+		for (int start = 0; start < urls.size(); start += maxUrls) {
+			int end = start + maxUrls;
+			if (end > urls.size()) {
+				end = urls.size();
+			}
+			StringBuilder sb = new StringBuilder((end - start) * LENGTH_OF_BASE_URL);

```

Comment:
Tracking LENGTH_OF_BASE_URL just to provide a good default to StringBuilder is probably not a good optimization. We should just keep the code clean and initialize StringBuilder with no arguments. (It'll amortize a few times, but you'll never notice.)